packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverBody/useTeachingPopoverBody.ts

```diff
@@ -1,34 +1,41 @@
 import * as React from 'react';
-import { getIntrinsicElementProps, slot } from '@fluentui/react-utilities';
+import { getIntrinsicElementProps, resolveShorthand } from '@fluentui/react-utilities';
 import type { TeachingPopoverBodyProps, TeachingPopoverBodyState } from './TeachingPopoverBody.types';
-
 /**
- * Create the state required to render TeachingPopoverBody.
- *
- * The returned state can be modified with hooks such as useTeachingPopoverBodyStyles_unstable,
- * before being passed to renderTeachingPopoverBody_unstable.
- *
- * @param props - props from this instance of TeachingPopoverBody
- * @param ref - reference to root HTMLDivElement of TeachingPopoverBody
+ * Returns the props and state required to render the component
+ * @param props - TeachingPopoverBody properties
+ * @param ref - reference to root HTMLElement of TeachingPopoverBody
  */
 export const useTeachingPopoverBody_unstable = (
   props: TeachingPopoverBodyProps,
-  ref: React.Ref<HTMLDivElement>,
+  ref: React.Ref<HTMLElement>,
 ): TeachingPopoverBodyState => {
+  const { as, media, mediaLength } = props;
+
+  let mediaShorthand;
+
+  // Image shouldn't be rendered if its src is not set
+  if (media?.src) {
+    mediaShorthand = resolveShorthand(media, {
```

Comment:
`resolveShorthand()` is deprecated API, please use `slot.*`

Reply:
Removed all use-cases of resolveShorthand - we still have instances of useARIAButtonShorthand - is there a replacement for that or is that non-deprecated?